src/Tests/NUnitTestExtractorTests/TestExtractorTests.cs

```diff
@@ -22,51 +24,21 @@ public static class TestExtractorTests
         [Test]
         public static void WritingToFile_InvalidDll_NothingWritten(string dll, string level)
         {
-            List<string> dlls = new List<string>();
-            dlls.Add(dll);
-            
-            StringBuilder builder = new StringBuilder();
+            NUnitTestExtractorApp.ScopeLevel = level;
 
-            InvokeGetTestsSimulatingFileOutput(builder, dlls, NUnitTestExtractorApp.ParseLevel(level));
+            var ex = Assert.Throws<FileNotFoundException>(() => NUnitTestExtractorApp.GenerateInfoFromAssembly(dll));
 
-            Assert.That(builder.ToString(), Is.Empty, "Nothing should be written.");
+            Assert.That(ex.Message, Is.StringContaining("File was not found:"));
         }
 
         [TestCaseSource(typeof(ValidDataTestCasesCollection))]
         [Description("Write data to a file using a valid dll and level. Verify the correct data is written.")]
         [Test]
         public static void WritingToFile_ValidDllAndEveryLevel_GetsWrittenWithProperFormat(string dll, string level)
         {
-            List<string> dlls = new List<string>();
-            dlls.Add(dll);
-
-            StringBuilder builder = new StringBuilder();
-
-            InvokeGetTestsSimulatingFileOutput(builder, dlls, NUnitTestExtractorApp.ParseLevel(level));
-
-            string builderText = builder.ToString();
-            string firstLine = builderText.Substring(0,builderText.IndexOf(Environment.NewLine, StringComparison.OrdinalIgnoreCase));
-
-            NUnitTestExtractorApp.Level myLevel = NUnitTestExtractorApp.ParseLevel(level);
-
-            Assert.That(VerifyOutputFormat(myLevel).IsMatch(firstLine), "Regex format should match output.");
-        }
-        
-        [TestCaseSource(typeof(ValidDataTestCasesCollection))]
-        [Description("Write data to a file using a valid dll and level. Verify the file contains no duplicates.")]
-        [Test]
-        public static void WritingToFile_ValidDllAndEveryLevel_FileContainsNoDuplicates(string dll, string level)
-        {
-            List<string> dlls = new List<string>();
-            dlls.Add(dll);
-
-            StringBuilder builder = new StringBuilder();
-            
-            InvokeGetTestsSimulatingFileOutput(builder, dlls, NUnitTestExtractorApp.ParseLevel(level));
+            NUnitTestExtractorApp.ScopeLevel = level;
 
-            string[] lines = builder.ToString().Split('\n');
-           
-            Assert.AreEqual(lines.Length, lines.Distinct().Count());
+            NUnitTestExtractorApp.GenerateInfoFromAssembly(dll);
```

Comment:
[Must] This test doesn't verify anything now.

Reply:
We need to revisit and re-write the Extractor utility tests completely.